v2/yarpcgrpc/handler.go

```diff
@@ -246,3 +246,32 @@ func handlerErrorToGRPCError(err error, mdWriter *metadataWriter) error {
 	}
 	return status.Error(grpcCode, message)
 }
+
+func handleResponse(
+	encoding yarpc.Encoding,
+	info *yarpcerror.Info,
+	resBuf *yarpc.Buffer,
+	serverStream grpc.ServerStream,
+	mdWriter *metadataWriter,
+) error {
+	// This is a regular response that we should send
+	if info == nil && resBuf != nil {
+		return serverStream.SendMsg(resBuf.Bytes())
+	}
+
+	// This is an application error
+	if info != nil {
+		if info.Name != "" {
+			mdWriter.AddSystemHeader(ErrorNameHeader, info.Name)
+		}
+		if resBuf != nil {
+			switch encoding {
+			case yarpcprotobuf.Encoding:
+				mdWriter.AddSystemHeader(ErrorDetailsHeader, resBuf.String())
+			case yarpcjson.Encoding, yarpcthrift.Encoding:
+				mdWriter.AddSystemHeader(ApplicationErrorHeader, resBuf.String())
```

Comment:
Can we now remove `ApplicationErrorHeaderValue` from `v2/yarpcgrpc/headers.go:81`? According to the comment above that, it says that `rpc-application-header`'s value represents an error if the value is non-empty. What definition does this refer to, yarpc is this a yarpc definition?

Reply:
Yea, it looks like this header value wasn't actually used anywhere except as a placeholder. On the outbound side, the client would only check if `ApplicationErrorHeader` exists. In v2, I'd like to use this header to actually hold the serialized error in JSON/Thrift's case.

I might be missing something, @AlexAPB, @kriskowal is this correct and is it safe for me to take this approach?

Reply:
for backwards compatibility, we'll need this header to be set whenever there's an application error. This also goes for the above protobuf statement.